haskellPackages: update stackage and hackage

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).
  
We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.
  
### haskellPackages Workflow Summary
  
Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).
  
The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer ha
s been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

---
    
This is the follow-up to #215925. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

dalpd committed:
haskellPackages.espial: Jailbreak to avoid upper bound on bytestring

Espial was broken because the version being tracked had a restrictive upper
bound on bytestring that isn't neccessary. Here we jailbreak after asserting
the version is 0.0.11 so that it fails once we there's a new release.

dalpd committed:
haskellPackages.espial: Add dalpd as the maintainer

dalpd committed:
haskellPackages.patat: Add dalpd as the maintainer

Commit:
haskellPackages.tdlib: patch for aeson >= 2.0 compat

Unbreak tdlib and its dependencies by same upstream (tdlib-types,
tdlib-gen, language-tl).

Co-authored-by: sternenseemann <sternenseemann@systemli.org>

sternenseemann committed:
all-cabal-hashes: 2023-02-13T17:53:53Z -> 2023-02-19T09:15:19Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

sternenseemann committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

sternenseemann committed:
haskellPackages.h-raylib, haskellPackages.bytepatch: unmark broken

Both packages had a new release fixing the issues we found after the
last bump, so they are easily fixed.

sternenseemann commented:
@ofborg build haskellPackages.Crypto

sternenseemann committed:
maintainers/haskell: generate core-pkgs hackage2nix configuration

`ghc-pkg list` tells us everything hackage2nix needs to know. In the
past the core-packages list and compiler setting in hackage2nix was
maintained manually which inevitably leads to it being forgot once in a
while – this will then mess with flag resolution when generating the
package set in some cases. Luckily, we can just let a simple derivation
do this for us.

Resolves #202621.

sternenseemann committed:
release-haskell.nix: don't try building known to fail js backends

aarch64-linux will continue failing unless we find another solution.
For x86_64-darwin we won't learn anything by it being a constaint star
in our build failure reports.

sternenseemann committed:
git-annex: update sha256 for 10.20230214

sternenseemann committed:
haskell.packages.*.[cC]abal*: use process 1.6.17.0 for GHC < 9.2.5

Cabal 3.8 and friends support process 1.6.17.0 which has recently been
released on Hackage, so we can fix the evaluation errors this way.

sternenseemann commented:
@ofborg build haskellPackages.bytepatch

raehik committed:
haskell.packages.ghc94.ormolu: fix eval by bumping version

maralorn committed:
maintainers/scripts/haskell/hydra-report: Add comments with error causes to broken list

cdepillabout committed:
Merge pull request #217010 from dalpd/dad-fix-espial

haskellPackages.espial: Use latest on GitHub (0.0.16) to unbreak

github-actions[bot] committed:
Merge master into haskell-updates

maralorn committed:
Merge pull request #217238 from raehik/ghc94-ormolu-pin-bump

 haskell.packages.ghc94.ormolu: fix eval by bumping version

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
Revert "haskellPackages.heystone: fix darwin dylibs"

This reverts commit 32eecc6cc96ba426cda9a8e7ab66d931eb141cee.

Reason for revert: This fix doesn't alleviate the problem in two ways:

1. It is in the aarch64-darwin section of configuration-darwin.nix,
   the problem appears on x86_64-darwin (bytepatch doesn't support
   aarch64).

2. When moving the override into the general *-darwin section, the
   problem with bytepatch is also unaffected, see my attempt in
   https://github.com/NixOS/nixpkgs/pull/217201.

sternenseemann committed:
haskellPackages.swisstable: "reflect" AVX2 requirement in platforms

github-actions[bot] committed:
Merge master into haskell-updates

mikesperber committed:
snappy: unbreak build

This just applies in the patches from a long-standing unmerged pull
request.

This transitively also unbreaks snappy-framing and snappy-lazy.

mikesperber committed:
haskell-tensorflow: unbreak build

This just moves the package to the current head in the official repo,
which has since seen patches that unbreak the ghc-9 build and work
with our version of TensorFlow.

iteratee committed:
haskell.compiler.ghc94*: backport Cabal bugfix for Paths_ modules

There is a code generation bug in Cabal-3.6.3.0. For packages configured with
--enable-relocatable, Cabal would generate code that doesn't compile.

There isn't an upstream issue, but the issue is described in the commit that
fixed it:
https://github.com/haskell/cabal/commit/6c796218c92f93c95e94d5ec2d077f6956f68e98

It was fixed in Cabal-3.8.*
Backport the fix to the Cabal library that ships with ghc-9.4.4

Cabal 3.8 ships with ghc-9.6, so when 9.6 is released this fix shouldn't be
necessary.

maralorn committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn committed:
Merge pull request #217812 from mikesperber/unbreak-snappy-haskell-tensorflow

Unbreak snappy and tensorflow Haskell packages

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.cabal2nix-unstable: 2023-02-15 -> 2023-02-25

Correctly resolves gmodule pkg-config name now:
https://github.com/NixOS/cabal2nix/commit/6c1dc8908ba4d3c81549b3131539d0ce289cb92c

vcunat commented:
I don't know the workflow of haskell updates, but could you add bump of `haskell-gi` to 0.26.3?  On current `staging-next` branch it's needed to build `haskellPackages.gi-harfbuz`.  I just tested manually
```diff
--- a/pkgs/development/haskell-modules/hackage-packages.nix
+++ b/pkgs/development/haskell-modules/hackage-packages.nix
@@ -129407,4 +129407,4 @@ self: {
        pname = "haskell-gi";
-       version = "0.26.2";
-       sha256 = "05r84czb05n69g7p7jazljh95yzdh2lpzgjjypgpg75mh83igr2w";
+       version = "0.26.3";
+       sha256 = "sha256-jsAb3JCSHCmi2dp9bpi/J3NRO/EQFB8ar4GpxAuBGOo=";
        setupHaskellDepends = [ base Cabal cabal-doctest ];
```

github-actions[bot] committed:
Merge master into haskell-updates

vcunat commented:
So, now that diff is in nixpkgs master as 1af0d67ebe1 &ndash; I thought that unbreaking until the next regeneration is an improvement.

_As `staging-next` merged, you'll see rebuild of everything, BTW._

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.haskell-gi: move temporary upgrade into overlay

This achieves the same as 1af0d67ebe17baaa14029379e430f21a3507a53b, but
prevents it being rolled back by package set regeneration.

sternenseemann committed:
haskellPackages.cabal2nix-unstable: 2023-02-25 -> 2023-02-27

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.compiler.ghc927: init at 9.2.7

https://www.haskell.org/ghc/blog/20230227-ghc-9.2.7-released.html

sternenseemann committed:
haskell-language-server: allow hls-call-hierarchy >= 1.2

Upstream change not released yet:
https://github.com/haskell/haskell-language-server/commit/05b248dfacc307c3397b334635cb38298aee9563

maralorn committed:
Merge pull request #217242 from maralorn/broken-reasons

maintainers/scripts/haskell/hydra-report: Add comments with error causes to broken list

sternenseemann committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1791397](https://hydra.nixos.org/eval/1791397) of nixpkgs commit [81d5cb1](https://github.com/NixOS/nixpkgs/commits/81d5cb1550ef0a58c5ee529c743065cc93a7fb64) as of 2023-02-28 23:50 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskellPackages.Cabal_3_8_1_0: apply revision 2

Requested in #218740.

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates
